cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -32,9 +32,9 @@ def vm_name():
 
 
 @pytest.fixture(scope="module")
-def test_vm(request, provider_init, provider_crud, provider_mgmt, vm_name):
+def test_vm(request, provider_init, provider_crud, provider_mgmt, provider_data, vm_name):
     """Fixture to provision appliance to the provider being tested if necessary"""
-    vm = Vm(vm_name, provider_crud)
+    vm = Vm(vm_name, provider_crud, template_name=provider_data['full_template'])
```

Comment:
If we don't have a `full_template` key in the provider data, it seems like this should uncollect.

Does this work like you'd expect?
```diff
diff --git a/cfme/tests/infrastructure/test_snapshot.py b/cfme/tests/infrastructure/test_snapshot.py
index ad76dbc..2975377 100644
--- a/cfme/tests/infrastructure/test_snapshot.py
+++ b/cfme/tests/infrastructure/test_snapshot.py
@@ -14,7 +14,7 @@ from utils.wait import wait_for
 pytestmark = [pytest.mark.long_running]
 
 
-pytest_generate_tests = testgen.generate(testgen.infra_providers, scope="module")
+pytest_generate_tests = testgen.generate(testgen.infra_providers, 'full_template', scope="module")
 
 
 @pytest.fixture(scope="module")
@@ -32,9 +32,9 @@ def vm_name():
 
 
 @pytest.fixture(scope="module")
-def test_vm(request, provider_init, provider_crud, provider_mgmt, provider_data, vm_name):
+def test_vm(request, provider_init, provider_crud, provider_mgmt, full_template, vm_name):
     """Fixture to provision appliance to the provider being tested if necessary"""
-    vm = Vm(vm_name, provider_crud, template_name=provider_data['full_template'])
+    vm = Vm(vm_name, provider_crud, full_template)
 
     if not provider_mgmt.does_vm_exist(vm_name):
         vm.create_on_provider(find_in_cfme=True)
```